insalan/tournament/admin.py

```diff
@@ -63,11 +64,72 @@
     ]),
 ]
 
+class SeatCanvasWidget(forms.Widget):
+    """
+    Custom widget for the seat canvas
+    """
+    def render(self, name, value, attrs=None, renderer=None):
+        return '<canvas id="seatCanvas" width="500" height="500" ></canvas>'
+
+class SeatCanvas(forms.Field):
+    """
+    Custom field for the seat canvas
+    """
+    widget = SeatCanvasWidget
+
+    def clean(self, value):
+        value = json.loads(value)
+        return value
+
+class EventForm(forms.ModelForm):
+    """
+    Custom form for the Event model
+    """
+    seats = SeatCanvas()
+    oldseats = forms.JSONField(widget=forms.HiddenInput, required=False)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        seats = Seat.objects.filter(event=self.instance)
+        self.fields["oldseats"].initial = [(seat.x, seat.y) for seat in seats]
+
+    def clean(self) -> dict[str, Any] | None:
+        seats = self.cleaned_data.get("seats")
+
+        if seats:
```

Comment:
If seats is an empty list ([]) the app will not go into the branch and no seats will be deleted because in python empty list... is false :upside_down_face:. You should use `seats is not None`